### PR TITLE
Azure Data Factory: enable AAD auth for Azure SQL DB and DW

### PR DESCRIPTION
**Pull request by xiaoyingLJ-zz**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process. 

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [x] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [x] If applicable, the PR references the bug/issue that it fixes.
- [x] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [x] My spec meets the review criteria:
  - [x] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [x] The spec follows the guidelines described in the [Swagger checklist](../documentation/swagger-checklist.md) document.
  - [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

File: specification/datafactory/resource-manager/Microsoft.DataFactory/preview/2017-09-01-preview/entityTypes/LinkedService.json [hunk 1]
```diff
@@ -95,6 +95,18 @@
...
+        "servicePrincipalId": {
+          "type": "object",
+          "description": "The ID of the service principal used to authenticate against Azure SQL Data Warehouse. Type: string (or Expression with resultType string)."
+        },
+        "servicePrincipalKey": {
+          "$ref": "../datafactory.json#/definitions/SecretBase",
+          "description": "The key of the service principal used to authenticate against Azure SQL Data Warehouse."
+        },
+        "tenant": {
+          "type": "object",
```
Comment (marstr) [inline specification/datafactory/resource-manager/Microsoft.DataFactory/preview/2017-09-01-preview/entityTypes/LinkedService.json]: We've been talking about this a lot lately. It seems like we actually don't have a great way to model this ATM. [Technically, a `string` is not a JSON `object`](json.org). However, the ability to use `oneOf` in your Swagger won't be available until we begin supporting Swagger v3 specification. There's a more thorough conversation happening here: https://github.com/Azure/autorest/issues/2827

At the moment, you're still doing the right thing, but I want to spread awareness of this problem.

**Review by marstr — APPROVED** [review]

**Comment by marstr** [comment]
These are small enough changes, and going into the `preview` folder, so I don't think that we need ARM review.

**Comment by marstr** [comment]
@xiaoyingLJ, are you ready for me to merge these changes?

**Comment by xiaoyingLJ** [comment]
Thanks Martin, we are ready to merge these changes.

**Pull request merged**